### DISCUSSION TITLE
Use xdotool api instead of run in fork shell for better three-finger dragging performance

### DISCUSSION THREAD

**Discussion by ferstar** [riley-martin/gestures#6] [Ideas]
I noticed that three-finger dragging triggers a lot of fork shells, which is very inefficient. So I used the xdotool api instead and got a really great performance boost(x10~100), especially with intentionally dragging a window quickly.

Here is my code: https://github.com/ferstar/gestures

I just merged code from: https://github.com/marsqing/libinput-three-finger-drag/blob/master/src/xdo_handler.rs 

into your base, and it works! lol~

**Comment by riley-martin** [comment]
I like that idea for the most part, but xdotool does not support Wayland, which I use, and is becoming more and more common. If there were rust bindings for something like [ydotool](https://github.com/ReimuNotMoe/ydotool), which works on X and Wayland, I would be interested in using that instead, but I have not found any.

I do not want to have separate code just for handling X and Wayland.
